archinstall/lib/installer.py

```diff
@@ -836,41 +835,40 @@ def _add_systemd_bootloader(
 					# We add in the default timeout to support dual-boot
 					loader_data[index] = line.removeprefix('#')
 
-		with loader_conf.open('w') as loader:
-			loader.writelines(loader_data)
+		loader_conf.write_text('\n'.join(loader_data) + '\n')
 
 		# Ensure that the $BOOT/loader/entries/ directory exists before we try to create files in it
 		entries_dir = loader_dir / 'entries'
 		entries_dir.mkdir(parents=True, exist_ok=True)
 
 		comments = (
-			'# Created by: archinstall\n',
-			f'# Created on: {self.init_time}\n'
+			'# Created by: archinstall',
+			'# Created on: ' + self.init_time
```

Comment:
I think I prefer to keep the f-string. Even if I know init_time is a strong it feels better knowing f-string will ensure its a str() object. nitpicking from my side.